Week 04/Lecture 07/Assignment 01/lecture_7/src/main/java/com/example/lecture_7/config/AppConfig.java

```diff
@@ -0,0 +1,34 @@
+package com.example.lecture_7.config;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+import com.example.lecture_7.EmployeeWork;
+import com.example.lecture_7.entity.Employee;
+
+@Configuration
+public class AppConfig {
+
+    @Bean
+    public EmployeeWork employeeWork() {
+        return new EmployeeWork();
+    }
+
+    // Constructor Injection
+    // uncomment this and comment the Setter Injection implementation to demo
+    /* @Bean
+    public Employee employee() {
```

Comment:
Can we inject employeeWork bean into employee bean via parameter ?

Reply:
Thank you for the suggestion.

I have modified the `employee` bean definition to use constructor injection for the `employeeWork` dependency. This ensures that the dependency is provided at the time of bean creation, which is generally a better practice in terms of immutability and clarity of dependencies.

The updated code is as follows:
```java
@Bean
public Employee employee(EmployeeWork employeeWork) {
    return new Employee("101", "John Doe", 30, employeeWork);
}
```

I have pushed the changes in the latest commit.